sources/Context.cpp

```diff
@@ -184,6 +216,43 @@ Context::exportOutputs() {
   diagramDirectory.append(basename_ + outputs::constants::diagramDirectorySuffix);
   outputs::Diagram diagramWriter(outputs::Diagram::DiagramDefinition(basename_, diagramDirectory.generic_string(), generators_, hvdcLineDefinitions_));
   diagramWriter.write();
+
+  if (def_.simulationKind == SimulationKind::SECURITY_ANALYSIS) {
+    if (validContingencies_) {
+      for (const auto& c : (*validContingencies_).get()) {
```

Comment:
Then validity must be checked in two different places:

- in the writers when building outputs for each valid contingency.
- in the simulation when adding a scenario to dynawo algorithms for every valid contingency.

Instead of checking validity twice, I preferred to build a list of references to valid ones.